source/usage-examples/findOne.go

```diff
@@ -0,0 +1,42 @@
+// This program finds a document titled 'The Room' in your MongoDB deployment.
+package main
+
+import (
+	"context"
+	"fmt"
+	"go.mongodb.org/mongo-driver/bson"
+	"go.mongodb.org/mongo-driver/mongo"
+	"go.mongodb.org/mongo-driver/mongo/options"
+	"log"
+)
+
+func main() {
+
+	// Replace the uri string with your MongoDB deployment's connection string.
+	uri := "mongodb+srv://<user>:<password>@<cluster-url>?retryWrites=true&w=majority"
+	ctx := context.TODO()
+	client, err := mongo.Connect(ctx, options.Client().ApplyURI(uri))
+	if err != nil {
+		panic(err)
+	}
+	// Disconnect the client once the function returns.
+	defer func() {
+		if err = client.Disconnect(ctx); err != nil {
+			panic(err)
+		}
+	}()
+	coll := client.Database("sample_mflix").Collection("movies")
+	var result bson.M
+	opts := options.FindOne()
+	opts = opts.SetProjection(bson.M{"_id": 0, "title": 1, "imdb": 1})
+	opts = opts.SetSort(bson.D{{"rating", -1}})
```

Comment:
Thanks for the note. You are correct, they do update the find one instance and do not need to assign to opts.

Interestingly, in testing this I found that the set sort method was incorrect, we need to sort on "imdb.rating" rather than "rating". The sort was ignored, and it didn't cause a problem because only one document in the sample collection has the title "The Room". 

Fixes for incorrect sort field:

Node PR: https://github.com/mongodb/docs-node/pull/161
Java PR: https://github.com/mongodb/docs-java/pull/74